transmogrifier/helpers.py

```diff
@@ -120,31 +118,29 @@ def validate_date_range(
     if start_date_object and end_date_object:
         if start_date_object <= end_date_object:
             return True
-        else:
-            logger.debug(
-                "Record ID '%s' has a later start date than end date: '%s', '%s'",
-                source_record_id,
-                start_date,
-                end_date,
-            )
-            return False
-    else:
         logger.debug(
-            "Record ID '%s' has invalid values in a date range: '%s', '%s'",
+            "Record ID '%s' has a later start date than end date: '%s', '%s'",
             source_record_id,
             start_date,
             end_date,
         )
         return False
+    logger.debug(
+        "Record ID '%s' has invalid values in a date range: '%s', '%s'",
+        source_record_id,
+        start_date,
+        end_date,
+    )
+    return False
 
 
-class DeletedRecord(Exception):
+class DeletedRecordError(Exception):
```

Comment:
I've been making this switch of trying to add "Error" as well since moving to Ruff.  Most of the time, I generally like it, but I wonder if it makes sense in this situation.

Reading a bit about python exceptions, I think there is case to be made that sometimes exceptions are used for **flow control**, which I feel is the case here.

I understand the linting suggestion to help make it clear that this class is an exception, and not get it confused for something else, which makes sense.  What about something like `DeletedRecordEvent`?  To me, the "Event" suffix still aligns with raising it, plus handling it,  but it distinguishes it from some kind of error.  My feeling is that we correctly understand that a record is deleted, but raising an exception is just a tidy way of sending an **event** that should be handled higher in the application logic and flow.

Reply:
I felt a little weird about adding `Error` this one but you summed up the reason why, I think `DeletedRecordEvent` is much more accurate